sdk/keyvault/Azure.Security.KeyVault.Keys/src/JsonWebKey.cs

```diff
@@ -255,14 +275,15 @@ public RSA ToRSA(bool includePrivateParameters = false)
 
             if (includePrivateParameters)
             {
-                var bitLength = rsaParameters.Modulus.Length * 8;
+                int byteLength = rsaParameters.Modulus.Length;
+                rsaParameters.D = ForceBufferLength(nameof(D), D, byteLength);
 
-                rsaParameters.D = ForceBufferLength(nameof(D), D, bitLength / 8);
-                rsaParameters.DP = ForceBufferLength(nameof(DP), DP, bitLength / 16);
-                rsaParameters.DQ = ForceBufferLength(nameof(DQ), DQ, bitLength / 16);
-                rsaParameters.P = ForceBufferLength(nameof(P), P, bitLength / 16);
-                rsaParameters.Q = ForceBufferLength(nameof(Q), Q, bitLength / 16);
-                rsaParameters.InverseQ = ForceBufferLength(nameof(QI), QI, bitLength / 16);
+                byteLength >>= 1;
```

Comment:
This probably could use some comments as to what our expectations of length are

Reply:
This is actually already documented / linked about 5 lines above what GitHub is showing,

```csharp
            // Key parameter length requirements defined by 2.2.2.9.1 RSA Private Key BLOB specification: https://docs.microsoft.com/openspecs/windows_protocols/ms-wcce/5cf2e6b9-3195-4f85-bc18-05b50e6d4e11
            var rsaParameters = new RSAParameters
            {
                Exponent = ForceBufferLength(nameof(E), E, 4),
                Modulus = TrimBuffer(N),
            };

            if (includePrivateParameters)
```